File: langs/cpp/testdata/fixture.cpp
#include <string>

namespace zoo {

class Animal {
public:
  std::string name;
  virtual void speak() {}
};

class Runnable {
public:
  virtual void run() {}
};

class Dog : public Animal, public Runnable {
public:
  int tricks;
  Dog() {}
  ~Dog() {}
  void bark();
};

template <typename T>
class Box : public Container<T> {
public:
  T value;
  void push(T v) {}
};

struct Point {
  int x;
  int y;
};

} // namespace zoo

enum class Color { RED, GREEN };

void zoo::Dog::bark() {}

void free_func(int a) {}

File: langs/java/testdata/fixture.java
package zoo;

interface Drawable {
    void draw();
}

interface Named {
    String name();
}

interface Entity extends Drawable, Named {
}

abstract class Animal implements Entity {
    protected String name;
    int legs = 4;

    Animal(String name) {
        this.name = name;
    }

    abstract void speak();

    class Collar {
        String color;
    }
}

class Dog extends Animal implements Comparable<Dog> {
    private int tricks;

    Dog(String name) {
        super(name);
    }

    @Override
    void speak() {
    }

    static Dog create(String name) {
        return new Dog(name);
    }
}

enum Color {
    RED, GREEN
}

record Point(int x, int y) {
}
